### PR TITLE
Fix Dynamodb connection

### PR DESCRIPTION
**Pull request by ayush-shah**
### Describe your changes:

Fixes Dynamodb listtables check
With valid Session token
image

with invalid session token
image

#
### Type of change:

- [x] Bug fix
- [x] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

**Review by OnkarVO7 — DISMISSED** [review]

File: ingestion/src/metadata/ingestion/source/database/dynamodb/connection.py [hunk 1]
```diff
@@ -32,6 +33,14 @@ def get_connection(connection: DynamoDBConnection):
     return AWSClient(connection.awsConfig).get_dynamo_client()
 
 
+def check_list_tables(client):
+    """
+    Test ListTables under dynamodb
+    """
+    for _ in client.tables.all():
+        return True
```
Comment (nahuelverdugo) [inline ingestion/src/metadata/ingestion/source/database/dynamodb/connection.py:41]: What are we trying to test here, if that call fails or not? Maybe something like this:
```suggestion
def check_list_tables(client) -> bool:
    """
    Test ListTables under dynamodb
    """
    try:
        client.tables.all()
    except Exception:
        return False
    return True
```

File: openmetadata-service/src/main/resources/json/data/testConnections/database/dynamodb.json [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
-    ]
-  }
\ No newline at end of file
+  "name": "DynamoDB",
+  "fullyQualifiedName": "DynamoDB",
+  "displayName": "DynamoDB Test Connection",
+  "description": "This Test Connection validates the access against the database and basic metadata extraction of schemas and tables.",
+  "steps": [
+    {
+      "name": "ListTables",
```
Comment (nahuelverdugo) [inline openmetadata-service/src/main/resources/json/data/testConnections/database/dynamodb.json:8]: Not 100% sure, do we need to migrate this or not?

File: ingestion/src/metadata/ingestion/source/database/dynamodb/connection.py (hunk 1 above)
Comment (ayush-shah) [reply ingestion/src/metadata/ingestion/source/database/dynamodb/connection.py:41]: so client.tables.all() passes even with wrong token, but when we run it as an iterator, it fails giving us a proper error.

File: openmetadata-service/src/main/resources/json/data/testConnections/database/dynamodb.json (hunk 2 above)
Comment (ayush-shah) [reply openmetadata-service/src/main/resources/json/data/testConnections/database/dynamodb.json:8]: we will need it

**Review by nahuelverdugo — DISMISSED** [review]

**Review by OnkarVO7 — APPROVED** [review]

**Review by OnkarVO7 — APPROVED** [review]

File: ingestion/src/metadata/ingestion/source/database/dynamodb/connection.py (hunk 1 above)
Comment (pmbrull) [reply ingestion/src/metadata/ingestion/source/database/dynamodb/connection.py:41]: @ayush-shah then IMO we should have this as a comment in the method

File: ingestion/src/metadata/ingestion/source/database/dynamodb/connection.py [hunk 3]
```diff
@@ -32,6 +33,14 @@ def get_connection(connection: DynamoDBConnection):
     return AWSClient(connection.awsConfig).get_dynamo_client()
 
 
+def check_list_tables(client):
+    """
+    Test ListTables under dynamodb
```
Comment (pmbrull) [inline ingestion/src/metadata/ingestion/source/database/dynamodb/connection.py:38]:
```suggestion
    Test ListTables under dynamodb
    
    `client.tables.all()` passes even with the wrong token, but when we run it as an iterator,
    it fails and gives us a proper error.
```

**Review by pmbrull — APPROVED** [review]

**Pull request merged**